shadow.c

```diff
@@ -607,6 +607,9 @@ static char *template_to_instance(char *filename, int options TSRMLS_DC)
 		if(SHADOW_G(debug) & SHADOW_DEBUG_PATHCHECK) fprintf(stderr, "In template: %s\n", filename);
 		if((options & OPT_CHECK_EXISTS) && shadow_cache_get(filename, &newname) == SUCCESS) {
 			if(SHADOW_G(debug) & SHADOW_DEBUG_PATHCHECK) fprintf(stderr, "Path check from cache: %s => %s\n", filename, newname);
+			if(realpath) {
+            			efree(realpath);
```

Comment:
It does implicitly through `estrndup`...

Makes me wonder if there are other leaks from this as well.